packages/react-server/src/ReactFizzServer.js

```diff
@@ -3857,7 +3857,6 @@ function abortTask(task: Task, request: Request, error: mixed): void {
   } else {
     boundary.pendingTasks--;
     if (boundary.status !== CLIENT_RENDERED) {
-      boundary.status = CLIENT_RENDERED;
```

Comment:
The point of setting this right away is to avoid a situation where something throws below and we're not in this status.

We really shouldn't throw below though.